src/Catalog/PackageCatalogItemCreator.cs

```diff
@@ -0,0 +1,240 @@
+                        {
+                            item = null;
+
+                            _telemetryService.TrackMetric(
+                                TelemetryConstants.BlobModified,
+                                metric: 1,
+                                properties: GetProperties(packageItem, blob));
+                        }
+                    }
+                }
+                else
+                {
+                    _telemetryService.TrackMetric(
+                        TelemetryConstants.NonExistentPackageHash,
+                        metric: 1,
+                        properties: GetProperties(packageItem, blob));
+                }
+            }
+
+            return item;
+        }
+
+        private async Task<PackageCatalogItem> GetPackageViaHttpAsync(FeedPackageDetails packageItem, DateTime timestamp, PackageCatalogItem item, CancellationToken cancellationToken)
+        {
+            // When downloading the package binary, add a query string parameter
+            // that corresponds to the operation's timestamp.
+            // This query string will ensure the package is not cached
+            // (e.g. on the CDN) and returns the "latest and greatest" package metadata.
+            var packageUri = Utilities.GetNugetCacheBustingUri(packageItem.ContentUri, timestamp.ToString("O"));
+            HttpResponseMessage response = null;
+
+            try
+            {
+                using (_telemetryService.TrackDuration(
+                    TelemetryConstants.PackageDownloadSeconds,
+                    new Dictionary<string, string>()
+                    {
+                        { TelemetryConstants.Id, packageItem.PackageId?.ToLowerInvariant() },
+                        { TelemetryConstants.Version, packageItem.PackageVersion?.ToLowerInvariant() },
+                    }))
+                {
+                    response = await _httpClient.GetAsync(packageUri, cancellationToken);
+                }
+            }
+            catch (TaskCanceledException tce)
+            {
+                // If the HTTP request timed out, a TaskCanceledException will be thrown.
+                throw new HttpClientTimeoutException($"HttpClient request timed out in {nameof(FeedHelpers.DownloadMetadata2Catalog)}.", tce);
+            }
+
+            if (response.IsSuccessStatusCode)
+            {
+                using (var stream = await response.Content.ReadAsStreamAsync())
+                {
+                    item = Utils.CreateCatalogItem(
+                        packageItem.ContentUri.ToString(),
+                        stream,
+                        packageItem.CreatedDate,
+                        packageItem.LastEditedDate,
+                        packageItem.PublishedDate);
+
+                    if (item == null)
+                    {
+                        _logger.LogWarning("Unable to extract metadata from: {PackageDetailsContentUri}", packageItem.ContentUri);
+                    }
```

Comment:
Is this the right behavior? The package will be in v2 but not v3.

Reply:
It is a non-goal as part of my V3 throughput work to make behavioral changes, especially not in this PR.  Please open a new issue if you think this should change.